web/modules/custom/dbcdk_community/src/Form/QuarantineCreateForm.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\dbcdk_community\Form\QuarantineCreateForm.
+ */
+
+namespace Drupal\dbcdk_community\Form;
+
+use DBCDK\CommunityServices\Api\QuarantineApi;
+use DBCDK\CommunityServices\ApiException;
+use DBCDK\CommunityServices\Model\Profile;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Create a Community Quarantine.
+ */
+class QuarantineCreateForm extends FormBase implements ContainerInjectionInterface {
+
+  /**
+   * The date format the Community Client expects.
+   */
+  const DATE_FORMAT = 'Y-m-d';
+
+  /**
+   * The Community Service Profile object from the username context.
+   *
+   * @var \DBCDK\CommunityServices\Model\Profile $profile
+   */
+  protected $profile;
+
+  /**
+   * The DBCDK Community Service Quarantine API.
+   *
+   * @var \DBCDK\CommunityServices\Api\QuarantineApi $quarantineApi
+   */
+  protected $quarantineApi;
+
+  /**
+   * QuarantineCreateForm constructor.
+   *
+   * @param \DBCDK\CommunityServices\Model\Profile $profile
+   *   The Community Profile we wish to grant a quarantine.
+   * @param \DBCDK\CommunityServices\Api\QuarantineApi $quarantine_api
+   *   The Community Service Quarantine API.
+   */
+  public function __construct(Profile $profile, QuarantineApi $quarantine_api) {
+    $this->profile = $profile;
+    $this->quarantineApi = $quarantine_api;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    try {
+      $profile_filter = [
+        'where' => [
+          'username' => $container->get('request_stack')->getCurrentRequest()->get('username'),
+        ],
+      ];
+      /* @var \DBCDK\CommunityServices\Api\ProfileApi */
+      $profile_api = $container->get('dbcdk_community.api.profile');
+      $profile = $profile_api->profileFindOne(json_encode($profile_filter));
+    }
+    catch (ApiException $e) {
+      \Drupal::logger('DBCDK Community Service')->error($e);
```

Comment:
I tend to retrieve the logger factory from the container and inject the logger instance into the class to reduce coupling.

What do you think about this?